libraries/plugins/stats_export/stats_export_plugin.cpp

```diff
@@ -0,0 +1,156 @@
+
+#include <steem/plugins/block_data_export/block_data_export_plugin.hpp>
+#include <steem/plugins/block_data_export/exportable_block_data.hpp>
+
+#include <steem/plugins/stats_export/stats_export_plugin.hpp>
+
+#include <steem/chain/account_object.hpp>
+#include <steem/chain/database.hpp>
+#include <steem/chain/global_property_object.hpp>
+#include <steem/chain/index.hpp>
+#include <steem/chain/operation_notification.hpp>
+
+#include <fstream>
+#include <iostream>
+#include <sstream>
+
+namespace steem { namespace plugins { namespace stats_export {
+
+using steem::chain::block_notification;
+using steem::chain::database;
+using steem::chain::dynamic_global_property_object;
+
+using steem::protocol::account_name_type;
+using steem::protocol::authority;
+using steem::protocol::signed_transaction;
+
+using steem::plugins::block_data_export::block_data_export_plugin;
+using steem::plugins::block_data_export::exportable_block_data;
+
+using steem::plugins::chain::chain_plugin;
+
+namespace detail {
+
+struct api_stats_transaction_data_object
+{
+   account_name_type     user;
+   uint32_t              size = 0;
+};
+
+class api_stats_export_data_object
+   : public exportable_block_data
+{
+   public:
+      api_stats_export_data_object() {}
+      virtual ~api_stats_export_data_object() {}
+
+      virtual void to_variant( fc::variant& v )const override
+      {
+         fc::to_variant( *this, v );
+      }
+
+      dynamic_global_property_object                        global_properties;
+      std::vector< api_stats_transaction_data_object >      transaction_stats;
+      uint64_t                                              free_memory = 0;
+};
+
+} } } }
+
+FC_REFLECT( steem::plugins::stats_export::detail::api_stats_transaction_data_object, (user)(size) )
+FC_REFLECT( steem::plugins::stats_export::detail::api_stats_export_data_object, (global_properties)(transaction_stats)(free_memory) )
+
+namespace steem { namespace plugins { namespace stats_export { namespace detail {
+
+class stats_export_plugin_impl
+{
+   public:
+      stats_export_plugin_impl( stats_export_plugin& _plugin ) :
+         _db( appbase::app().get_plugin< chain_plugin >().db() ),
+         _self( _plugin ),
+         _export_plugin( appbase::app().get_plugin< block_data_export_plugin >() )
```

Comment:
Because you are using the `block_data_export_plugin` as a dependency, you should specify it in `APPBASE_PLUGIN_REQUIRES`